cgt_calc/parsers/trading212.py

```diff
@@ -98,32 +105,57 @@ def __init__(self, header: list[str], row_raw: list[str], filename: str):
         self.price_foreign = decimal_or_none(row["Price / share"])
         self.currency_foreign = row["Currency (Price / share)"]
         self.exchange_rate = decimal_or_none(row["Exchange rate"])
-        self.transaction_fee = decimal_or_none(row.get("Transaction fee (GBP)", "0"))
-        self.finra_fee = decimal_or_none(row.get("Finra fee (GBP)", "0"))
-        self.stamp_duty = decimal_or_none(row.get("Stamp duty (GBP)", "0"))
-        self.conversion_fee = decimal_or_none(
-            row.get("Currency conversion fee (GBP)", "0")
-        )
-        fees = (
-            (self.transaction_fee or Decimal(0))
-            + (self.finra_fee or Decimal(0))
-            + (self.conversion_fee or Decimal(0))
-        )
+        self.transaction_fee = Decimal(row.get("Transaction fee (GBP)") or "0")
+        transaction_fee_foreign = Decimal(row.get("Transaction fee") or "0")
+        if transaction_fee_foreign > 0:
+            if row.get("Currency (Transaction fee)") != "GBP":
+                raise ParsingError(
+                    filename,
+                    "The transaction fee is not in GBP which is not supported yet",
+                )
+            self.transaction_fee += transaction_fee_foreign
+        self.finra_fee = Decimal(row.get("Finra fee (GBP)") or "0")
+        self.stamp_duty = Decimal(row.get("Stamp duty (GBP)") or "0")
+        self.conversion_fee = Decimal(row.get("Currency conversion fee (GBP)") or "0")
+        conversion_fee_foreign = Decimal(row.get("Currency conversion fee") or "0")
+        if conversion_fee_foreign > 0:
+            if row.get("Currency (Currency conversion fee)") != "GBP":
+                raise ParsingError(
+                    filename,
+                    "The transaction fee is not in GBP which is not supported yet",
+                )
+            self.conversion_fee += conversion_fee_foreign
+        fees = self.transaction_fee + self.finra_fee + self.conversion_fee
         if "Total" in row:
             amount = decimal_or_none(row["Total"])
             currency = row["Currency (Total)"]
         else:
             amount = decimal_or_none(row["Total (GBP)"])
             currency = "GBP"
+        if (
+            amount is not None
+            and (action == ActionType.BUY or self.raw_action == "Withdrawal")
+            and amount > 0
+        ):
+            amount *= -1
         price = (
-            abs(amount / quantity)
+            abs((amount + fees) / quantity)
```

Comment:
If amount is negative but fees are always positive?
```suggestion
            (abs(amount) + fees) / quantity)
```

Reply:
No, that won't work. The formula is to remove the fees from the total paid/received.
- If purchasing 1 share costs £10 with £1 fee, price / share is abs(-10+1)/1 = £9
- If selling 1 share gives me £9 with £1 fee, price / share is abs(9+1)/1 = £10